### ISSUE TITLE
Feature request: Flowchart node size or emphasis

### ISSUE THREAD

**Issue by pineapplemachine** [mermaid-js/mermaid#3183]
**Is your feature request related to a problem? Please describe.**

Problem: There is no documented way to make some flowchart nodes more visually emphasized than others. Applying `height` styling results in misaligned arrows and misaligned text. Applying `font-size` or `font-weight` styling results in misaligned text, and truncated text when the larger text exceeds the size of the node shape.

**Describe the solution you'd like**

Ideally, every node would accept a percentage size factor that scales both the shape and the font by that amount, perhaps as a class style.

Practically, charts should probably just be more responsive to changes in height and font styling.

Or at the very least, a flag to make certain nodes emphasized would go a long way. These nodes should be roughly 150% the height and font size of other nodes.

**Describe alternatives you've considered**

Current styling options cannot effectively achieve this.

**Additional context**

![image](https://user-images.githubusercontent.com/7266412/176213475-968c6d6c-2e10-4879-aaf1-338ed9f5128d.png)

```
graph TD
    A[Very Important Node]:::em1 --> B
    B[Less Important Node] --> C[Another Important Node]:::em2
    D[Also Important Node]:::em3 --> B
    C --> E[Important Node Redux]:::em4
    classDef em1 font-size:20px
    classDef em2 height:75px
    classDef em3 font-weight:bold
    classDef em4 font-size:20px,height:75px,font-weight:bold
```

Potentially related issues:

https://github.com/mermaid-js/mermaid/issues/3032
https://github.com/mermaid-js/mermaid/issues/2139
https://github.com/mermaid-js/mermaid/issues/1540
https://github.com/mermaid-js/mermaid/issues/1250

**Comment by knsv** [comment]
Good idea! I think it needs to be done before the styling but it could be done!

**Comment by octokatherine** [comment]
I also am wanting to make the font size of a node bigger but am having issues with text being truncated

**Comment by pranavm2109** [comment]
Hi, I'd like to work on this issue!

**Comment by pranavm2109** [comment]
Quick clarification: if the font-size is set to 2em, then the box height dynamically readjusts itself, without having the need to introduce any additional emphasis parameter like the one that is requested. For instance, in the example code, if the height attribute for em4 in  `classDef em4 font-size:20px,height:75px,font-weight:bold` is removed, then the box resizes itself without any truncation of text. So, could you please clarify the problem?

**Comment by pineapplemachine** [comment]
> Quick clarification: if the font-size is set to 2em, then the box height dynamically readjusts itself, without having the need to introduce any additional emphasis parameter like the one that is requested. For instance, in the example code, if the height attribute for em4 in `classDef em4 font-size:20px,height:75px,font-weight:bold` is removed, then the box resizes itself without any truncation of text. So, could you please clarify the problem?

The layout behavior seems to have changed since I originally created the issue. The layout is drastically different now than the previously shared screenshot in more ways than this. I haven't been following this project and I have no idea, I'm just here because I still follow this one issue, but it certainly _looks_ like this was already fixed at some point in the last year and a half:

![image](https://github.com/user-attachments/assets/a7b95734-579c-4907-a0b3-86b461f1e588)

**Comment by pranavm2109** [comment]
@jgreywolf It seems like according to @pineapplemachine , this issue has been resolved somewhere in the last year and a half. If that is the case, could this issue be closed?

**Comment by popzip** [comment]
From Sean on Linkedin:

I wish we could have multiple font sizes! Have one use cas of a big massive graph and it would be easier to format if there was ability to control font sizes across different types of objects
